data/build_probe_data.py

```diff
@@ -0,0 +1,120 @@
+import tensorflow.compat.v1 as tf
+from tqdm import tqdm
+from collections import defaultdict
+import sklearn
+import numpy as np
+import json
+
+with tf.io.gfile.GFile("gs://e2e_central/data/ml-sequences-train.tsv", 'r') as f:
+  sequence_list = list(f)
+  data = []
+  for sequence_str in tqdm(sequence_list):
+    data.append([x.strip() for x in sequence_str.replace("\n", "").replace('\t', "").split("@") if x.strip() != ""])
+
+with tf.io.gfile.GFile("gs://e2e_central/data/ml-sequences-test.tsv", 'r') as f:
+  sequence_list = list(f)
+  for sequence_str in tqdm(sequence_list):
+    data.append([x.strip() for x in sequence_str.replace("\n", "").replace('\t', "").split("@") if x.strip() != ""])
+
+# def write_json(filepath, dictionary):
+#   with tf.io.gfile.GFile(filepath, 'w') as f:
+#     json.dump(dictionary, filepath)
+
+# def write_json(filepath, dictionary):
+#   with tf.io.gfile.GFile(filepath, 'w') as f:
+#     json.dump(dictionary, filepath)
+
+
+
+
+movie_set = set()
+popularity = defaultdict(int)
+
+for seq in data:
+  for movie in seq:
+    movie_set.add(movie)
+    popularity[movie] += 1
+
+# for seq in data:
+#   if len(set(seq)) != 10:
+#     print(seq)
+
+
+num_sequences = len(data)
+
+popular_movies = list(sorted(movie_set, key=lambda x: popularity[x], reverse=True))
+movie_set = sorted(movie_set)
+vocab_size = len(movie_set)
+embed = dict(zip(movie_set, list(range(vocab_size))))
+unembed = dict(zip(list(range(vocab_size)), movie_set))
+
+movie_ids = {
+  "all_movies": movie_set,
+  "movie_count": vocab_size,
+  "movie_to_id": embed,
+  "id_to_movie": unembed,
+  "popularity": popularity
+}
+with tf.io.gfile.GFile("gs://e2e_central/data/probes/movie_id_info.json", 'w') as f:
+  json.dump(movie_ids, f)
+
+
+
+def create_cooccurrence(sequences):
```

Comment:
could we have all functions definitions at the top.
also please add a line docstring describing the function.
https://google.github.io/styleguide/pyguide.html 
Section 3.8.3